### PR TITLE
[REVIEW] [develop] Trello-WuVtjwgt: [DB] Error handling via Mongoose middleware

### PR DESCRIPTION
**Pull request by AndreySurzhan**
[Trello-WuVtjwgt](https://trello.com/c/WuVtjwgt): [DB] Error handling via Mongoose middleware
- [x] Rebase

### CODE REVIEW

File: utils/mongoosePlugins/notFoundErrorHandler.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            if (!doc) {
+                return next(error);
+            }
+
+            return next();
+        });
+    });
+};
+
+module.exports = notFoundErrorHandler;
```
Comment (AndreySurzhan) [inline utils/mongoosePlugins/notFoundErrorHandler.js]: additional line

File: utils/mongoosePlugins/notFoundErrorHandler.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+
```
Comment (AndreySurzhan) [inline utils/mongoosePlugins/notFoundErrorHandler.js]: extra line

File: package.json [hunk 3]
```diff
@@ -5,10 +5,11 @@
     "async": "^2.0.0-rc.6",
     "body-parser": "1.0.1",
     "crypto": "0.0.3",
-    "database": "3.6.13",
     "express": "4.0.0",
-    "mongoose": "3.6.13",
     "mongodb": "^2.1.18",
+    "mongoose": "^4.5.3",
```
Comment (AndreySurzhan) [inline package.json]: remove mask

File: controllers/project.js [hunk 4]
```diff
@@ -311,46 +233,26 @@ var projectController = {
...
-                if (!validation.isOwner(data.userId, project.owners)) {
-                    error = {
-                        status: 403,
-                        message: 'User doesn\'t have permissions for updating project name: ' + data.userId
-                    };
-
+            .exec(function (error, doc) {
+                if (error) {
                     logger.error(error);
+                    logger.error('Project main currency was not updated: ' + data.id);
```
Comment (AndreySurzhan) [inline controllers/project.js]: rename

File: controllers/project.js [hunk 5]
```diff
@@ -1,12 +1,11 @@
 /// Libs
 var Logger = require('../libs/log');
 var Q = require('q');
-var validation = require('../utils/validation');
+var mongoose = require('mongoose');
```
Comment (AndreySurzhan) [inline controllers/project.js]: extra

**Comment by AndreySurzhan** [comment]
@wildeastengineer please review

File: controllers/project.js [hunk 6]
```diff
@@ -245,57 +195,28 @@ var projectController = {
...
-                deferred.reject(error);
-            });
 
+        ProjectModel.findOneAndUpdate({
+            _id: data.id,
+            owners: data.userId
+        }, {
+            mainCurrency: data.mainCurrency
+        }, {
+            runValidators: true
```
Comment (wildeastengineer) [inline controllers/project.js]: Why it is here but not in places before

**Pull request merged**